### PR TITLE
Add integration test run_on_script_test

### PR DESCRIPTION
**Pull request by SinaChavoshi**

### CODE REVIEW

File: tensorflow_cloud/python/tests/integration/call_run_on_script_with_keras_ctl_test.py [hunk 1]
```diff
@@ -11,14 +11,37 @@
...
+class TensorflowCloudOnScriptTest(tf.test.TestCase):
+    def setUp(self):
+        super(TensorflowCloudOnScriptTest, self).setUp()
+        self.test_data_path = os.path.join(
+            os.path.dirname(os.path.abspath(__file__)), "../testdata/"
+        )
+
+    @mock.patch.object(sys, "exit", autospec=True)
+    def test_MWMS_on_script(self, mock_exit):
+        mock_exit.side_effect = RuntimeError("exit called")
```
Comment (ucdmkt) [inline tensorflow_cloud/python/tests/integration/call_run_on_script_with_keras_ctl_test.py:32]: This looks too hacky. 

asserting that sys.exit was called with 0 without raising any runtime error is a good enough start.

Comment (SinaChavoshi) [reply tensorflow_cloud/python/tests/integration/call_run_on_script_with_keras_ctl_test.py]: Done.

File: tensorflow_cloud/python/tests/integration/on_script_call_test.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+# The project id to use to run tests.
+# PROJECT_ID = os.environ['PROJECT_ID']
+#
+# The GCP region in which the end-to-end test is run.
+# REGION = os.environ['REGION']
+#
+# Unique ID for this build, can be used as a label for an AI Platform training job.
+# BUILD_ID = os.environ['BUILD_ID']
+
+class TensorflowCloudOnScriptTest(tf.test.TestCase):
```
Comment (pavithrasv) [inline tensorflow_cloud/python/tests/integration/on_script_call_test.py:37]: Can we rename this to RunOnScriptTest?

File: tensorflow_cloud/python/tests/integration/on_script_call_test.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
+# Copyright 2020 Google LLC. All Rights Reserved.
```
Comment (pavithrasv) [inline tensorflow_cloud/python/tests/integration/on_script_call_test.py:1]: Can we rename file to run_on_script_test.py?

File: tensorflow_cloud/python/tests/integration/on_script_call_test.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+"""Integration tests for calling tfc.run on a script with keras."""
+
+import os
+import sys
+from typing import Text
+from unittest import mock
+import tensorflow as tf
+import tensorflow_cloud as tfc
+
+# Following are the env variables available in test infrastructure:
```
Comment (pavithrasv) [inline tensorflow_cloud/python/tests/integration/on_script_call_test.py:23]: Thank you for the comments!

File: tensorflow_cloud/python/tests/integration/on_script_call_test.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        for folder in self.test_folders:
+            self.delete_dir(folder)
+
+    def delete_dir(self, path: Text) -> None:
+        """Deletes a directory if exists."""
+        if tf.io.gfile.isdir(path):
+            tf.io.gfile.rmtree(path)
+      
+    @mock.patch.object(sys, "exit", autospec=True)
+    def test_MWMS_on_script(self, mock_exit):
```
Comment (pavithrasv) [inline tensorflow_cloud/python/tests/integration/on_script_call_test.py:58]: Can we rename this to test_none_dist_strat_mwms_on_script?

File: tensorflow_cloud/python/tests/integration/on_script_call_test.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        tfc.run(
+            entry_point=os.path.join(
+                self.test_data_path, "mnist_example_using_ctl.py"
+            ),
+            distribution_strategy=None,
+            worker_count=1,
```
Comment (pavithrasv) [inline tensorflow_cloud/python/tests/integration/on_script_call_test.py:64]: Can we change this to 2?

File: tensorflow_cloud/python/tests/integration/on_script_call_test.py (hunk 2 above)
Comment (SinaChavoshi) [reply tensorflow_cloud/python/tests/integration/on_script_call_test.py:37]: done

File: tensorflow_cloud/python/tests/integration/on_script_call_test.py (hunk 3 above)
Comment (SinaChavoshi) [reply tensorflow_cloud/python/tests/integration/on_script_call_test.py:1]: done

File: tensorflow_cloud/python/tests/integration/on_script_call_test.py (hunk 5 above)
Comment (SinaChavoshi) [reply tensorflow_cloud/python/tests/integration/on_script_call_test.py:58]: done

File: tensorflow_cloud/python/tests/integration/on_script_call_test.py (hunk 6 above)
Comment (SinaChavoshi) [reply tensorflow_cloud/python/tests/integration/on_script_call_test.py]: done

File: tensorflow_cloud/python/tests/integration/on_script_call_test.py (hunk 4 above)
Comment (SinaChavoshi) [reply tensorflow_cloud/python/tests/integration/on_script_call_test.py]: :)

**Comment by SinaChavoshi** [comment]
FYI the integration tests are failing, though I believe it is a valid failure. TF 2.3.0 does not have a corresponding docker image yet ( it was released today ) 

```console
Traceback (most recent call last):
  File "/usr/lib/python3.5/unittest/mock.py", line 1157, in patched
    return func(*args, **keywargs)
  File "/tmpfs/SOURCE_ENV/tensorflow_cloud/src/python/tensorflow_cloud/tests/integration/run_on_script_test.py", line 66, in test_none_dist_strat_mwms_on_script
    self.test_data_path, "requirements.txt"),
  File "/tmpfs/BUILD_ENV/lib/python3.5/site-packages/tensorflow_cloud/core/run.py", line 222, in run
    docker_img_uri = container_builder.get_docker_image()
  File "/tmpfs/BUILD_ENV/lib/python3.5/site-packages/tensorflow_cloud/core/containerize.py", line 298, in get_docker_image
    self._get_tar_file_path()
  File "/tmpfs/BUILD_ENV/lib/python3.5/site-packages/tensorflow_cloud/core/containerize.py", line 123, in _get_tar_file_path
    self._create_docker_file()
  File "/tmpfs/BUILD_ENV/lib/python3.5/site-packages/tensorflow_cloud/core/containerize.py", line 170, in _create_docker_file
    "with an other TF version.".format(VERSION)
ValueError: There is no docker base image corresponding to the local TF version: 2.3.0. Please provide docker_base_image or try with an other TF version.
```

**Pull request merged**